File: realtime/android/webrtc/app/src/main/java/ai/inworld/voiceagent/state/TranscriptItem.kt
package ai.inworld.voiceagent.state

import java.util.UUID

enum class Role { User, Agent }

data class TranscriptItem(
    val id: String = UUID.randomUUID().toString(),
    val role: Role,
    val text: String,
    val isStreaming: Boolean = false,
)

File: realtime/flutter/webrtc/lib/state/transcript_item.dart
enum TranscriptRole { user, agent }

class TranscriptItem {
  TranscriptItem({
    required this.id,
    required this.role,
    required this.text,
    this.isStreaming = false,
  });

  final int id;
  final TranscriptRole role;
  String text;
  bool isStreaming;
}
